WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/config/AnalogComplicationConfigRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,754 @@
+
+                // Retrieve config item (contains data to populate view).
+                val backgroundComplicationConfigItem: BackgroundComplicationConfigItem =
+                    configItemType as BackgroundComplicationConfigItem
+
+                // Retrieve specific icons, strings, etc. needed for view holder.
+                val backgroundIconResourceId: Int = backgroundComplicationConfigItem.iconResourceId
+                val backgroundName: String = backgroundComplicationConfigItem.name
+
+                // Set icons, strings, etc. in view holder.
+                backgroundComplicationViewHolder.setIcon(backgroundIconResourceId)
+                backgroundComplicationViewHolder.setName(backgroundName)
+            }
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        val configItemType: ConfigItemType = settingsDataSet[position]
+        return configItemType.configType
+    }
+
+    override fun getItemCount(): Int {
+        return settingsDataSet.size
+    }
+
+    /** Updates the selected complication id saved earlier with the new information.  */
+    fun updateSelectedComplication(complicationProviderInfo: ComplicationProviderInfo?) {
+        Log.d(TAG, "updateSelectedComplication: $previewAndComplicationsViewHolder")
+
+        // Checks if view is inflated and complication id is valid.
+        if (selectedComplicationId >= 0) {
+            previewAndComplicationsViewHolder.updateComplicationViews(
+                selectedComplicationId,
+                complicationProviderInfo
+            )
+        }
+    }
+
+    override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView)
+        // Required to release retriever for active complication data on detach.
+        providerInfoRetriever.release()
+    }
+
+    fun updatePreviewColors() {
+        Log.d(TAG, "updatePreviewColors(): $previewAndComplicationsViewHolder")
+        previewAndComplicationsViewHolder.updateWatchFaceColors()
+    }
+
+    /**
+     * Displays watch face preview along with complication locations. Allows user to tap on the
+     * complication they want to change and preview updates dynamically.
+     */
+    inner class PreviewAndComplicationsViewHolder(view: View) :
+        RecyclerView.ViewHolder(view), View.OnClickListener {
+
+        private val watchFaceArmsAndTicksView: View
+        private val watchFaceHighlightPreviewView: View
+        private val watchFaceBackgroundPreviewImageView: ImageView
+
+        private val leftComplicationBackground: ImageView
+        private val rightComplicationBackground: ImageView
+        private val leftComplication: ImageButton
+        private val rightComplication: ImageButton
+        private var defaultComplicationDrawable: Drawable? = null
+        private var backgroundComplicationEnabled = false
+
+        override fun onClick(view: View) {
```

Comment:
The click handling should be handled by the activity, not by the adapter.
In the adapter constructor you should pass in a lambda for item clicked and here you just trigger that lambda